arbor/mc_cell_group.cpp

```diff
@@ -135,21 +85,54 @@ void mc_cell_group::advance(epoch ep, time_type dt, const event_lane_subrange& e
 
     PE(advance_eventsetup);
     staged_events_.clear();
+
+    cell_size_type ev_begin = 0, ev_mid = 0, ev_end = 0;
     // skip event binning if empty lanes are passed
     if (event_lanes.size()) {
-        for (auto lid: util::count_along(gids_)) {
-            auto& lane = event_lanes[perm_gids_[lid]];
+
+        std::vector<cell_size_type> idx_sorted_by_intdom(cell_to_intdom_.size());
+        cell_size_type n = 0;
+        std::generate(idx_sorted_by_intdom.begin(), idx_sorted_by_intdom.end(), [&]{ return n++; });
+        std::sort(idx_sorted_by_intdom.begin(), idx_sorted_by_intdom.end(),
+            [&](cell_size_type a, cell_size_type b) {
+            return cell_to_intdom_[a] < cell_to_intdom_[b];
+        });
+
+        int prev_intdom = -1;
```

Comment:
There may well be as many intdoms as there are cells in the group, so for consistency/documentation it would be better to use either `cell_size_type` to match the lane counting, or `fvm_index_type` to match the `cell_to_intdom_` mapping.

Reply:
Changed to fvm_index_type